### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by matthewpersicowork** [koalaman/shellcheck#2153]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

THEDIR=/tmp/o/b/b

for i in "$THEDIR"/*
do
    echo "$i"
done
    
```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:

I have two issues:

#### Issue #1
I got to this point because shellcheck threw an SC2045 at me (Iterating over ls output is fragile. Use globs.) when I was using
```
for i in $(ls "$THEDIR"/*)
...
```
and told me to use a glob. However a glob does not work on an empty dir. To wit:
```
$ THEDIR=o/b/b

$ ls -la $THEDIR
total 0
drwxrwxr-x 2 mpersico5 users 17 Feb 19 14:44 .
drwxrwxr-x 3 mpersico5 users 15 Feb 19 14:36 ..
-rw-rw-r-- 1 mpersico5 users  0 Feb 19 14:44 foo

$ for i in "$THEDIR"/*; do     echo "$i"; done
o/b/b/foo

$ rm o/b/b/foo
rm: remove regular empty file 'o/b/b/foo'? y

$ for i in "$THEDIR"/*; do echo "$i"; done
o/b/b/*
```
I want the last result to be empty, not an unresolved glob string. I think the glob advice is potentially dangerous.

**Comment by brother** [comment]
Dupe of #2154 ?

> I have two issues

incorrect paste or something where issue number 2 is missing or something.

**Comment by matthewpersicowork** [comment]
Dupe of #2154
